core/src/main/scala/org/apache/spark/status/AppStatusUtils.scala

```diff
@@ -17,16 +17,23 @@
 
 package org.apache.spark.status
 
-import org.apache.spark.status.api.v1.{TaskData, TaskMetrics}
+import org.apache.spark.status.api.v1.TaskData
 
 private[spark] object AppStatusUtils {
 
+  private val TASK_FINISHED_STATES = Set("FAILED", "KILLED", "SUCCESS")
+
+  private def isTaskFinished(task: TaskData): Boolean = {
+    TASK_FINISHED_STATES.contains(task.status)
+  }
+
   def schedulerDelay(task: TaskData): Long = {
-    if (task.taskMetrics.isDefined && task.duration.isDefined) {
+    if (isTaskFinished(task) && task.taskMetrics.isDefined && task.duration.isDefined) {
```

Comment:
`task.duration.isDefined` should be redundant now, right?

(I remember the duration didn't use to be set for running tasks, so this code worked, but apparently it changed while I worked on these changes...)

Reply:
Logically `duration` should be set for running tasks, to indicate how long a task has been run.

I feel it's safer to keep `task.duration.isDefined`, as we call `task.duration.get` below.